File: julia8024/PGS_42747.py

def solution(citations):
    answer = 0
    citations.sort(reverse=True)
    
    # a번 이상 인용된 논문이 b편 이상 -> a, b 중 작은 값이 H-index
    for i in range(len(citations)):
        tmp = citations[i] if citations[i] < i+1 else i+1
        # 방금 탐색한 H-index가 이전에 탐색한 H-index보다 큰 경우 방금 탐색한 값으로 변경
        answer = tmp if tmp > answer else answer
    
    return answer

File: youbbin/PGS_42747.java
import java.util.*;
public class PGS_42747 {
    // 코드 1 (2024.03.26)
//    public int solution(int[] citations) {
//        int[] hIdxArr = new int[citations.length];
//        for(int i = 0; i < citations.length; i++){
//            int h = 0;
//            for(int j = 0; j < citations.length; j++){
//                if(citations[i] <= citations[j]){
//                    h++;
//                }
//                if(h > citations[i]){
//                    h = 0;
//                }
//            }
//            hIdxArr[i] = h;
//        }
//
//        Arrays.sort(hIdxArr);
//        return hIdxArr[(citations.length)-1];
//
//    }

    // 코드 2 (2024.04.01)
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        for(int i = 0; i < citations.length; i++){
            int h = citations.length - i;
            if(citations[i] >= h){
                answer = h;
                break;
            }
        }
        return answer;
    }
}
